File: Swift/Structural/Bridge/Bridge/CSVEngine.swift
//
//  CSVEngine.swift
//  Bridge
//
//  Created by Anton on 26.01.17.
//  Copyright © 2017 None. All rights reserved.
//

import Foundation

class CSVEngine : ReportEngineProtocol {
    func buildReport(videos: [Video]) -> String {
        var output = "Id, Title, Url\n"
        
        for video in videos {
            output += String(format: "%@, '%@', %@\n", video.id, video.title, video.url)
        }
        
        return output
    }
}

File: Go/Structural/Bridge/ReportEngines/csvEngine.go
package ReportEngines

import (
	"github.com/MastersAcademy/ma_ap_2016/Go/Structural/Bridge/Models"
	"bytes"
	"fmt"
)

type CSVEngine struct {

}

func NewCSVEngine() ReportEngine {
	return &CSVEngine{}
}

func (e *CSVEngine) BuildReport(videos []Models.Video) string  {
	var buffer bytes.Buffer
	buffer.WriteString("Id, Title, Url\n")

	for _, v := range videos {
		buffer.WriteString(fmt.Sprintf("%v, '%s', %s\n", v.ID, v.Title, v.URL))
	}

	return buffer.String()
}
